google-ads-examples/src/main/java/com/google/ads/googleads/examples/errorhandling/HandleRateExceededError.java

```diff
@@ -0,0 +1,244 @@
+
+/**
+ * Handle RateExceededError in your application. To trigger the rate exceeded error, this code
+ * example runs 500 threads in parallel, each thread attempting to validate 100 keywords in a single
+ * request. Note that spawning 100 parallel threads is for illustrative purposes only, you shouldn't
+ * do this in your application.
+ */
+public class HandleRateExceededError {
+  private static class HandleRateExceededErrorParams extends CodeSampleParams {
+
+    @Parameter(names = ArgumentNames.CUSTOMER_ID, required = true)
+    private Long customerId;
+
+    @Parameter(names = ArgumentNames.AD_GROUP_ID, required = true)
+    private Long adGroupId;
+  }
+
+  public static void main(String[] args) throws InterruptedException {
+    HandleRateExceededErrorParams params = new HandleRateExceededErrorParams();
+    if (!params.parseArguments(args)) {
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerId = Long.parseLong("INSERT_CUSTOMER_ID_HERE");
+      params.adGroupId = Long.parseLong("INSERT_AD_GROUP_ID_HERE");
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new HandleRateExceededError()
+          .runExample(googleAdsClient, params.customerId, params.adGroupId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param adGroupId the ID of the ad group to which keywords are added.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, long customerId, long adGroupId)
+      throws InterruptedException {
+    final int NUM_THREADS = 500;
+
+    List<Thread> threads = new ArrayList<>();
+
+    // Adds 500 threads that validate keywords to a list.
+    for (int i = 0; i < NUM_THREADS; i++) {
```

Comment:
I'm concerned about an example that intentionally tries to exceed rate limits. Could you change this to just demonstrate the error handling/retry approach, but without the intentional triggering of limits?